streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractPartitionGroup.java

```diff
@@ -25,7 +25,29 @@
 
 abstract class AbstractPartitionGroup {
 
-    abstract boolean readyToProcess(long wallClockTime);
+    /**
+     * Result of readyToProcess check, containing both the readiness status
+     * and an optional diagnostic log message.
+     */
+    static final class ReadyToProcessResult {
+        private final boolean ready;
+        private final Optional<String> logMessage;
+
+        ReadyToProcessResult(final boolean ready, final Optional<String> logMessage) {
+            this.ready = ready;
+            this.logMessage = logMessage;
```

Comment:
Should we add a check, if `ready == true`, `logMessage` cannot be empty, and if `ready == true`, `logMessage` should be empty?